src/System.Private.CoreLib/shared/System/Text/ASCIIEncoding.cs

```diff
@@ -137,10 +175,14 @@ public override unsafe int GetByteCount(ReadOnlySpan<char> chars)
             if (byteIndex < 0 || byteIndex > bytes.Length)
                 throw new ArgumentOutOfRangeException(nameof(byteIndex), SR.ArgumentOutOfRange_Index);
 
-            int byteCount = bytes.Length - byteIndex;
+            // It's ok for either of these pointers to be null.
+            // The workhorse will handle it appropriately.
 
-            fixed (char* pChars = chars) fixed (byte* pBytes = &MemoryMarshal.GetReference((Span<byte>)bytes))
```

Comment:
>MemoryMarshal.GetReference [](start = 64, length = 26)

I assume the perf for regular pinning is not degrading from using MemoryMarshal.GetReference. right?

Reply:
`MemoryMarshal.GetReference` is used internally when we don't care whether the pointer is null. In our case, the workhorse routines can handle null pointers just fine, so we don't need the fixup that normal pinning provides.